packages/fuels/tests/contracts.rs

```diff
@@ -1331,6 +1336,119 @@ async fn low_level_call() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(not(feature = "fuel-core-lib"), feature = "rocksdb"))]
+#[test]
+fn db_rocksdb() {
+    use std::{fs, str::FromStr};
+
+    use fuels::{
+        accounts::wallet::WalletUnlocked,
+        client::{PageDirection, PaginationRequest},
+        crypto::SecretKey,
+        prelude::{setup_test_provider, DbType, Error, ViewOnlyAccount, DEFAULT_COIN_AMOUNT},
+    };
+
+    let temp_dir = tempfile::tempdir()
+        .expect("failed to make tempdir")
+        .into_path();
```

Comment:
This will stop the temp dir being deleted upon its handle being dropped. In cases of the test failing we might not execute the manual cleanup at the end of the test.

I suggest the handle is kept and its ownership moved into the future so that it may be dropped no matter how the test finishes.